frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts

```diff
@@ -1,10 +1,36 @@
 import { K8sResourceKind } from '@console/internal/module/k8s';
-import { NO_PROVISIONER, OCS_INTERNAL_CR_NAME, CEPH_STORAGE_NAMESPACE } from '../../constants';
+import {
+  NO_PROVISIONER,
+  OCS_INTERNAL_CR_NAME,
+  CEPH_STORAGE_NAMESPACE,
+  OCS_DEVICE_SET_REPLICA,
+} from '../../constants';
 
-export const getOCSRequestData = (
-  scName?: string,
-  storage?: string,
-  infra?: string,
+export const createDeviceSet = (scName: string, osdSize: string, portable: boolean): DeviceSet => ({
+  name: `ocs-deviceset-${scName}`,
+  count: 1,
+  portable,
+  replica: OCS_DEVICE_SET_REPLICA,
+  resources: {},
+  placement: {},
+  dataPVCTemplate: {
+    spec: {
+      storageClassName: scName,
+      accessModes: ['ReadWriteOnce'],
+      volumeMode: 'Block',
+      resources: {
+        requests: {
+          storage: osdSize,
+        },
+      },
+    },
+  },
+});
+
+export const createInternalAndAttachedStorageCluster = (
```

Comment:
This function is basically returning the request data after considering all the conditions and not creating a storage cluster. Hence it was named as `getOCSRequestData`. You can replace OCS word if you want to be very specific which is not necessary IMO. But naming this like createInternalAndAttachedStorageCluster doesn't fit here.

Reply:
Its not a generic function We would make it generic for all internal, external and attached devices.
It is used only for attached devices and internal
But something not needed now for this PR @gnehapk